pkg/restic/repository_manager.go

```diff
@@ -209,142 +159,59 @@ func (rm *repositoryManager) NewRestorer(ctx context.Context, restore *arkv1api.
-		if len(prefix) <= 1 {
-			continue
-		}
-
-		// strip the trailing '/' if it exists
-		repos = append(repos, strings.TrimSuffix(prefix, "/"))
-	}
-
-	return repos, nil
-}
-
-func (rm *repositoryManager) CheckAllRepos() error {
-	repos, err := rm.getAllRepos()
-	if err != nil {
-		return err
-	}
-
-	var eg sync.ErrorGroup
-	for _, repo := range repos {
-		this := repo
-		eg.Go(func() error {
-			rm.log.WithField("repo", this).Debugf("Checking repo %s", this)
-			return rm.CheckRepo(this)
-		})
-	}
-
-	return kerrs.NewAggregate(eg.Wait())
-}
-
-func (rm *repositoryManager) PruneAllRepos() error {
-	repos, err := rm.getAllRepos()
-	if err != nil {
-		return err
-	}
-
-	var eg sync.ErrorGroup
-	for _, repo := range repos {
-		this := repo
-		eg.Go(func() error {
-			rm.log.WithField("repo", this).Debugf("Pre-prune checking repo %s", this)
-			if err := rm.CheckRepo(this); err != nil {
-				return err
-			}
-
-			rm.log.WithField("repo", this).Debugf("Pruning repo %s", this)
-			if err := rm.PruneRepo(this); err != nil {
-				return err
-			}
-
-			rm.log.WithField("repo", this).Debugf("Post-prune checking repo %s", this)
-			return rm.CheckRepo(this)
-		})
-	}
-
-	return kerrs.NewAggregate(eg.Wait())
+	return rm.exec(InitCommand(rm.repoPrefix, name))
 }
 
 func (rm *repositoryManager) CheckRepo(name string) error {
 	rm.repoLocker.LockExclusive(name)
 	defer rm.repoLocker.UnlockExclusive(name)
 
-	cmd := CheckCommand(rm.config.repoPrefix, name)
+	cmd := CheckCommand(rm.repoPrefix, name)
 
-	return errorOnly(rm.exec(cmd))
+	return rm.exec(cmd)
 }
 
 func (rm *repositoryManager) PruneRepo(name string) error {
 	rm.repoLocker.LockExclusive(name)
 	defer rm.repoLocker.UnlockExclusive(name)
 
-	cmd := PruneCommand(rm.config.repoPrefix, name)
+	cmd := PruneCommand(rm.repoPrefix, name)
 
-	return errorOnly(rm.exec(cmd))
+	return rm.exec(cmd)
 }
 
 func (rm *repositoryManager) Forget(snapshot SnapshotIdentifier) error {
 	rm.repoLocker.LockExclusive(snapshot.Repo)
 	defer rm.repoLocker.UnlockExclusive(snapshot.Repo)
 
-	cmd := ForgetCommand(rm.config.repoPrefix, snapshot.Repo, snapshot.SnapshotID)
+	cmd := ForgetCommand(rm.repoPrefix, snapshot.Repo, snapshot.SnapshotID)
 
-	return errorOnly(rm.exec(cmd))
+	return rm.exec(cmd)
 }
 
-func (rm *repositoryManager) exec(cmd *Command) ([]byte, error) {
+func (rm *repositoryManager) exec(cmd *Command) error {
 	file, err := TempCredentialsFile(rm.secretsLister, cmd.Repo)
 	if err != nil {
-		return nil, err
+		return err
 	}
 	// ignore error since there's nothing we can do and it's a temp file.
 	defer os.Remove(file)
 
 	cmd.PasswordFile = file
 
-	output, err := cmd.Cmd().Output()
-	rm.log.WithField("repository", cmd.Repo).Debugf("Ran restic command=%q, output=%s", cmd.String(), output)
+	stdout, stderr, err := arkexec.RunCommand(cmd.Cmd())
+	rm.log.WithFields(logrus.Fields{
+		"repository": cmd.Repo,
+		"stdout":     stdout,
+		"stderr":     stderr,
+	}).Debugf("Ran restic command=%s", cmd.String())
```

Comment:
Sorry, I missed that `command` was also here - I'd recommend moving it to a structured field too. Trying to get 100% of our variable output to structured fields.